app/assets/javascripts/host_edit.js

```diff
@@ -893,17 +893,30 @@ function selectRelatedNetwork(subnetElement) {
   }
 
   var selected = null;
+  // prefer a match of the vlanid bounded by non digits
+  // this prevent vlanId=1 from matching "vlan100"
+  var vlanidregex = new RegExp(`(^|\\D)(${vlanId})($|\\D)`)
```

Comment:
I think we shouldn't use template in asset pipeline, as it works only since JS ES6 and we still support older platforms. And asset pipeline doesn't transpile through babel.
```suggestion
  var vlanidregex = new RegExp("(^|\\D)" + vlanId + "($|\\D)");
```